### PR TITLE
Refactor message_receiver to not use async tasks on message consuming

### PR DESCRIPTION
**Pull request by TarasLevelUp**
This will remove some "not consumed" coroutines.

### CODE REVIEW

File: src/asynqp/channel.py [hunk 1]
```diff
@@ -319,36 +329,32 @@ def receive_return(self, frame):
...
-    @asyncio.coroutine
     def receive_header(self, frame):
-        self.synchroniser.notify(frames.ContentHeaderFrame)
+        assert self.message_builder is not None, "Reveived unexpected header"
         self.message_builder.set_header(frame.payload)
         self.reader.ready()
 
-    @asyncio.coroutine
     def receive_body(self, frame):
+        assert self.message_builder is not None, "Reveived unexpected body"
```
Comment (benjamin-hodgson) [inline src/asynqp/channel.py]: Typo! Should be 'received' :wink:

File: src/asynqp/channel.py [hunk 2]
```diff
@@ -304,11 +311,14 @@ def receive_deliver(self, frame):
             payload.routing_key,
             payload.consumer_tag
         )
-        self.reader.ready()
 
-        yield from self.async_receive()
+        # Deliver the message to consumers when done
+        def callback(tag_msg):
+            self.consumers.deliver(*tag_msg)
+            self.reader.ready()
```
Comment (benjamin-hodgson) [inline src/asynqp/channel.py]: These callbacks don't seem to access the closure so there's no need for them to be nested. I'd prefer to see it as proper (private) method on the `MessageReceiver`.

I understand that it dispatches to a different function depending on whether it's a basic get or a basic deliver ([line 300](https://github.com/benjamin-hodgson/asynqp/pull/50/files#diff-a12f879848a4250e1a7b48cb26ff7fe5R300)). I find that slightly confusing. I'd prefer to have a Boolean property called something like `is_deliver`.

**Comment by benjamin-hodgson** [comment]
Looks good. I think those async methods are a hangover from before I brought in the actor stuff. Just a couple of minor things and then we can merge this :wink:

**Comment by Drizzt1991** [comment]
Fixed

**Pull request merged**